packages/circuits/circuits/preventDoubleAction.circom

```diff
@@ -0,0 +1,81 @@
+pragma circom 2.0.0;
+
+include "./circomlib/circuits/poseidon.circom";
+include "./circomlib/circuits/bitify.circom";
+include "./incrementalMerkleTree.circom";
+include "./epochKeyLite.circom";
+include "./leafHasher.circom";
+include "./identity.circom";
+
+template PreventDoubleAction(STATE_TREE_DEPTH, EPOCH_KEY_NONCE_PER_EPOCH, FIELD_COUNT, EPK_R) {
+
+    // Global state tree
+    signal input state_tree_indexes[STATE_TREE_DEPTH];
+    signal input state_tree_elements[STATE_TREE_DEPTH];
+
+    // Global state tree leaf: Identity & user state root
+    signal input identity_secret;
```

Comment:
Don't accept the identity secret as an input, use the calculated value from the identity nullifier/trapdoor. The `IdentityCommitment` component has a property `secret` that you can use.

Reply:
what would be the appropriate value for `genEpochKey: (identitySecret: bigint, attesterId: bigint | string, epoch: bigint | number, nonce: bigint | number)`'s first argument in the test? would that be `id.secretHash`?

Reply:
yes

Reply:
Oh I didn't notice that. That is erroneous, you should be using `identity_nullifier` to calculate the `identity_commitment`. The commitment never changes so the secret should never change. The [semaphore circuit](https://github.com/semaphore-protocol/semaphore/blob/main/packages/circuits/semaphore.circom#L47) is a good reference.

Reply:
That makes sense! Thank you for the reference.